### PR TITLE
Made 3 improvements.  First, set timeout to 15 seconds so it does not…

### PR DESCRIPTION
**Pull request by hitchingsh**
… get stuck.  Limit text returned to 2K per result.  Set time filter to last 30 days for web pages indexed within last 30 days.  These changes improve the information returned and answers given by ParlAI.

### CODE REVIEW

File: search_server.py [hunk 1]
```diff
@@ -201,7 +207,7 @@ def search(self, q: str, n: int) -> Generator[str, None, None]:
 
 class GoogleSearchServer(SearchABC):
     def search(self, q: str, n: int) -> Generator[str, None, None]:
-        return googlesearch.search(q, num=n, stop=None, pause=_DELAY_SEARCH)
+        return googlesearch.search(q, num=n, stop=None, pause=_DELAY_SEARCH, tbs="qdr:m")
```
Comment (JulesGM) [inline search_server.py:210]: can you remove `tbs="qdr:m"` please, we don't want pages from just this month to show up

Comment (JulesGM) [inline search_server.py]: https://github.com/MarioVilas/googlesearch/blob/master/googlesearch/__init__.py#L221
Is it indexed in the last 30 days or published in the last 30 days?

File: search_server.py [hunk 2]
```diff
@@ -161,6 +162,11 @@ def do_POST(self):
                     f"   {rich.markup.escape(maybe_content['url'])}"
                     # f"Content: {len(maybe_content['content'])}",
                 )
+
+                # Truncate text 
+                if len(maybe_content) > _TRUNCATE_TEXT_BYTES:
```
Comment (JulesGM) [inline search_server.py]: the `if` is redundant, just leave the next line please (if _TRUNCATE_TEXT_BYTES is too large, it just gets the whole array)

File: search_server.py [hunk 3]
```diff
@@ -29,7 +29,8 @@
 _STYLE_SKIP = ""
 _CLOSE_STYLE_GOOD = "[/]" if _STYLE_GOOD else ""
 _CLOSE_STYLE_SKIP = "[/]" if _STYLE_SKIP else ""
-
+_DEFAULT_URL_REQUEST_TIMEOUT = 15 # seconds
+_TRUNCATE_TEXT_BYTES = 2048
```
Comment (JulesGM) [inline search_server.py]: How did you get to the number 2048?

File: search_server.py [hunk 4]
```diff
@@ -29,7 +29,8 @@
 _STYLE_SKIP = ""
 _CLOSE_STYLE_GOOD = "[/]" if _STYLE_GOOD else ""
 _CLOSE_STYLE_SKIP = "[/]" if _STYLE_SKIP else ""
-
+_DEFAULT_URL_REQUEST_TIMEOUT = 15 # seconds
```
Comment (JulesGM) [inline search_server.py]: Good point about this. Should likely be much shorter actually, like 5 seconds. 
Why default? maybe change to `_REQUESTS_GET_TIMEOUT`

File: search_server.py (hunk 2 above)
Comment (JulesGM) [reply search_server.py]: Removing the `if` would also make `_TRUNCATE_TEXT_BYTES` being `None` disable the truncation, which would be good

**Review by JulesGM — COMMENTED** [review]
Thanks for the contribution, I have made a few comments

File: search_server.py (hunk 3 above)
Comment (JulesGM) [reply search_server.py]: I would rather you make `_TRUNCATE_TEXT_BYTES` a command-line parameter for `serve`, with a default value. Whether that value should be `None` or some number .. I'm not convinced yet.

Comment (JulesGM) [reply search_server.py]: I guess `_DEFAULT_URL_REQUEST_TIMEOUT` should likely also be a parameter with a default value, but that is less important

**Comment by hitchingsh** [comment]
Will submit a new merge request for new changes

**Pull request closed without merging**